File: src/com/wix/rt/cli/RTSettings.java
package com.wix.rt.cli;

import com.wix.rt.settings.Settings;
import org.jetbrains.annotations.NotNull;

/**
 * RT settings
 * Created by idok on 11/17/14.
 */
public class RTSettings {
    public String node;
    public String rtExecutablePath;
    public String config;
    public String cwd;
    public String targetFile;
    public String targetVersion;
    public String modules;
    public boolean dryRun;
    public boolean reactNative;

    public static RTSettings build(@NotNull String cwd, @NotNull String nodeInterpreter, @NotNull String rtBin, @NotNull String path, @NotNull String modules, boolean dryRun, boolean reactNative) {
        RTSettings settings = new RTSettings();
        settings.cwd = cwd;
        settings.rtExecutablePath = rtBin;
        settings.node = nodeInterpreter;
        settings.targetFile = path;
        settings.modules = modules;
        settings.dryRun = dryRun;
        settings.reactNative = reactNative;
        return settings;
    }

    public static RTSettings build(Settings settings) {
        RTSettings s = new RTSettings();
        s.rtExecutablePath = settings.rtExecutable;
        s.node = settings.nodeInterpreter;
        s.modules = settings.modules;
        s.reactNative = settings.reactNative;
        s.targetVersion = settings.targetVersion;
        return s;
    }

    public static RTSettings build(@NotNull Settings settings, @NotNull String cwd, @NotNull String path) {
        RTSettings s = build(settings);
        s.cwd = cwd;
        s.targetFile = path;
        return s;
    }

    public static RTSettings build(@NotNull String cwd, @NotNull String nodeInterpreter, @NotNull String rtBin) {
        return build(cwd, nodeInterpreter, rtBin, "", "", false, false);
    }
}

File: src/com/wix/rtk/cli/RTSettings.kt
package com.wix.rtk.cli

import com.wix.rt.settings.Settings

/**
 * RT settings
 * Created by idok on 11/17/14.
 */
data class RTSettings(val node: String, val rtExe: String, val cwd: String, val config: String = "", val targetFile: String = "",
                      val targetVersion: String = "", val modules: String = "", val dryRun: Boolean = false, val reactNative: Boolean = false) {
    companion object {
        @JvmStatic
        fun build(cwd: String, nodeInterpreter: String, rtBin: String, path: String = "", modules: String = "", dryRun: Boolean = false, reactNative: Boolean = false): RTSettings =
                RTSettings(nodeInterpreter, rtBin, "", cwd, path, "", modules, dryRun, reactNative)

        @JvmStatic
        fun buildVersion(cwd: String, nodeInterpreter: String, rtBin: String): RTSettings = RTSettings(nodeInterpreter, rtBin, cwd)

        @JvmStatic
        fun buildSettings(settings: Settings, cwd: String, path: String): RTSettings =
                RTSettings(settings.nodeInterpreter, settings.rtExecutable, cwd, "", path, settings.targetVersion, settings.modules, false, settings.reactNative)
    }
}

data class NodeExe(val node: String, val exe: String, val cwd: String? = null)

//class RTVersion(node: String, exe: String, cwd: String? = null, val x:String) : NodeExe(node, exe, cwd)
